lldb/source/DataFormatters/TypeSynthetic.cpp

```diff
@@ -67,7 +67,9 @@ TypeFilterImpl::FrontEnd::GetIndexOfChildWithName(ConstString name) {
       }
     }
   }
-  return UINT32_MAX;
+  return llvm::createStringError(
+      "'SyntheticChildrenFrontEnd::FrontEnd' cannot find index of child '%s'",
```

Comment:
```suggestion
      "'TypeFilterImpl::FrontEnd' cannot find index of child '%s'",
```

Reply:
I would suggest leaving out the function name entirely.

Reply:
@Michael137 My thinking is that these error messages are meant to be read by LLDB users, and the function name is a tool to aid in debugging LLDB itself. Users won't understand the meaning of the function name here, so we shouldn't have it in the user-facing error message. If this is useful information needed to debug LLDB, then we should log it instead.
Let me know if you think that's not a good idea!

Reply:
Hmm fair question. Adding logging to all these places feels a bit noisy. In my opinion its helpful to be as specific as possible, and if we wanted to present something more generic to the user eventually then we can change the error message at the point where we are about to present it. I'm not sure where these error messages will currently show up actually

Reply:
> Adding logging to all these places feels a bit noisy.

Currently the error is dropped silently. You wouldn't replace that with adding logging everywhere. My suggestion would be to:
- return errors that are understandable to users
- where necessary add logging in the error case with additional information

Reply:
> I'm not sure where these error messages will currently show up actually

The error messages in the Expected return results of CompilerType are surfaced when formatting ValueObjects, for example.

Reply:
Hmm my main concern is that we end up getting user reports with a very generic error and then have no way to really tell where we went wrong. Then adding logging after-the-fact is a pretty slow turnaround. I guess usually these types of issues require a reproducer to debug anyway, so maybe this isn't a valid concern? Wdyt? I don't have a strong opinion on this. But it would be easier/more convenient to go from non-generic error back to generic error than the other way around if we really need to